indexd/index/drivers/single_table_alchemy.py

```diff
@@ -0,0 +1,1539 @@
+
+    def delete(self, guid, rev):
+        """
+        Removes record if stored by backend.
+        """
+        with self.session as session:
+            query = session.query(Record)
+            query = query.filter(Record.guid == guid)
+
+            try:
+                record = query.one()
+            except NoResultFound:
+                raise NoRecordFound("no record found")
+            except MultipleResultsFound:
+                raise MultipleRecordsFound("multiple records found")
+
+            if rev != record.rev:
+                raise RevisionMismatch("revision mismatch")
+
+            auth.authorize("delete", [u.resource for u in record.authz])
+
+            session.delete(record)
+
+    def add_version(
+        self,
+        current_guid,
+        form,
+        new_did=None,
+        size=None,
+        file_name=None,
+        metadata=None,
+        urls_metadata=None,
+        version=None,
+        urls=None,
+        acl=None,
+        authz=None,
+        hashes=None,
+        description=None,
+        content_created_date=None,
+        content_updated_date=None,
+    ):
+        """
+        Add a record version given guid
+        """
+        urls = urls or []
+        acl = acl or []
+        authz = authz or []
+        hashes = hashes or {}
+        metadata = metadata or {}
+        urls_metadata = urls_metadata or {}
+
+        with self.session as session:
+            query = session.query(Record).filter_by(guid=current_guid)
+
+            try:
+                record = query.one()
+            except NoResultFound:
+                raise NoRecordFound("no record found")
+            except MultipleResultsFound:
+                raise MultipleRecordsFound("multiple records found")
+
+            auth.authorize("update", [u for u in record.authz] + authz)
+
+            baseid = record.baseid
+            record = Record()
+            guid = new_did
+            if not guid:
+                guid = str(uuid.uuid4())
+                if self.config.get("PREPEND_PREFIX"):
+                    guid = self.config["DEFAULT_PREFIX"] + guid
+
+            record.guid = guid
+            record.baseid = baseid
+            record.rev = str(uuid.uuid4())[:8]
+            record.form = form
+            record.size = size
+            record.file_name = file_name
+            record.version = version
+            record.description = description
+            record.urls = urls
+            record.acl = acl
+            record.authz = authz
+            record.hashes = hashes
+            record.record_metadata = metadata
+
+            self._validate_and_format_content_dates(
+                record=record,
+                content_created_date=content_created_date,
+                content_updated_date=content_updated_date,
+            )
+            record.content_created_date = content_created_date
+            record.content_updated_date = content_updated_date
```

Comment:
```suggestion
```
it looks like `_validate_and_format_content_dates` sets the right values, so we shouldn't overwrite them